backend/workflow_manager/endpoint/source.py

```diff
@@ -119,25 +123,22 @@ def valid_file_patterns(self, required_patterns: list[Any]) -> list[str]:
                 wildcard.extend(patterns.get(pattern, []))
         return wildcard
 
-    def list_file_from_api_storage(self) -> list[str]:
+    def list_file_from_api_storage(
+        self, file_hashes: dict[str, FileHash]
+    ) -> tuple[dict[str, FileHash], int]:
         """List all files from the api_storage_dir directory."""
-        files: list[str] = []
-        if not self.api_storage_dir:
-            return files
-        for file in os.listdir(self.api_storage_dir):
-            file_path = os.path.join(self.api_storage_dir, file)
-            if os.path.isfile(file_path):
-                files.append(file_path)
-        return files
-
-    def list_files_from_file_connector(self) -> list[str]:
+        return file_hashes, len(file_hashes)
```

Comment:
@vishnuszipstack kind of confused why we even have this function anymore. Looks like we are simply returning the input with it's length. Is this even required?

Also the initial description of the method and it's name doesn't looks like to match it's functionality anymore. Can't we just remove this function?

Reply:
@ritwik-g removed